packages/lab/src/Flow/Node/BaseNode.tsx

```diff
@@ -231,21 +231,37 @@ export const HvFlowBaseNode = ({
           </div>
 
           <div className={classes.inputsContainer}>
-            {inputs?.map((input) => (
-              <div className={classes.inputContainer} key={input.id}>
-                <Handle
-                  type="target"
-                  isConnectableStart={false}
-                  id={input.id}
-                  position={Position.Left}
-                />
-                <HvTypography>{input.label}</HvTypography>
-                {input.isMandatory &&
-                  !isInputConnected(id, "target", input.id!, inputEdges) && (
-                    <div className={classes.mandatory} />
-                  )}
-              </div>
-            ))}
+            {inputs?.map((input, idx) => {
+              const handleId = input.id ?? idx.toString();
+              const isConnected = isInputConnected(
+                id,
+                "target",
+                handleId,
+                inputEdges
+              );
+              return (
+                <div className={classes.inputContainer} key={input.id}>
+                  <Handle
+                    type="target"
+                    isConnectableStart={false}
+                    id={input.id}
+                    position={Position.Left}
+                    {...(isConnected && {
+                      style: {
+                        backgroundColor: theme.colors.secondary_60,
+                        width: 8,
+                        height: 8,
+                      },
```

Comment:
What do you think about creating a class for this to have all the styling in the same place? (same for the outputs below)